### PR TITLE
Move KVErrors to participant-state-kv-errors [KVL-1371]

### PR DESCRIPTION
**Pull request by hubert-da**
I did run into a circular dependency, so had to create a new package.

It is a prerequisite to move `kvutils` to the private repo.

### CODE REVIEW

**Review by nicu-da — APPROVED** [review]

File: ledger/participant-state-errors/src/main/scala/com/daml/ledger/participant/state/kv/errors/KVErrors.scala [hunk 1]
```diff
@@ -15,6 +13,8 @@ import com.daml.error.{
   Resolution,
 }
 
+import java.time.Instant
+
 @Explanation(
```
Comment (hubert-da) [inline ledger/participant-state-errors/src/main/scala/com/daml/ledger/participant/state/kv/errors/KVErrors.scala:18]: @pbatko-da, is the documentation going to be generated after this move? How to check it?

File: ledger/participant-state-errors/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
```
Comment (miklos-da) [inline ledger/participant-state-errors/BUILD.bazel:1]: Nit:
To signal that this is a KV specific module, you might want to name it as `participant-state-kv-errors`.
Moving these error definitions to under a common root with the other error documentation related definitions such as `com.daml.ledger.error.definitions.kv.*` may also make sense (however, then the module name/location would need to change as well).

File: ledger/participant-state-errors/src/main/scala/com/daml/ledger/participant/state/kv/errors/KVErrors.scala [hunk 3]
```diff
@@ -1,11 +1,9 @@
 // Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
```
Comment (miklos-da) [inline ledger/participant-state-errors/src/main/scala/com/daml/ledger/participant/state/kv/errors/KVErrors.scala:1]: Nit:
The naming of this class is incorrect, as it should rather be `KvErrors` to follow [correct camel-casing of abbreviations](https://docs.scala-lang.org/style/naming-conventions.html).

**Comment by hubert-da** [comment]
After this commit, the documentation stopped generating properly 😕

**Comment by hubert-da** [comment]
Ok, had to regenerate the docs because of `KVErrors` -> `KvErrors`.

**Review by fabiotudone-da — APPROVED** [review]

**Pull request merged**